### ISSUE TITLE
one more problem with trivial f.p. groups

### ISSUE THREAD

**Issue by ThomasBreuer** [gap-system/gap#4738]
The following happens in the master branch as well as in GAP 4.11.1.
```
gap> F:= FreeGroup( 1 );;                                        
gap> G:= F / [ F.1 ];;
gap> Size( G );
1
gap> PcGroupFpGroup( G );
Error, the rewriting system must be confluent at /.../lib/rwsgrp.gi:328 called from
GroupByRws( col ) at /.../lib/rwspcgrp.gi:1173 called from
[...]
you can 'return;' to continue
```
In this situation, `GroupByRws` has called `IsConfluent` on its argument `col`.
The two relevant methods for `IsConfluent` ("method for finite polycylic rewriting systems" and "generic method for polycyclic rewriting systems") give different answers.
```
brk> meth:= ApplicableMethod( IsConfluent, [ col ], 1, "all" );;
brk> meth[1]( col );
false
brk> meth[2]( col );
true
```
Which of them is right?

**Comment by ThomasBreuer** [comment]
Update: The same happens not only for trivial f.p. groups but also when just some generator is the identity:
```
gap> F:= FreeGroup( 2 );;
gap> rels:= [ One(F), F.2^2, F.1, Comm( F.2, F.1 ) ];;
gap> G:= F / rels;;
gap> PcGroupFpGroup( G );
Error, the rewriting system must be confluent at /.../lib/rwsgrp.gi:325 called from
[...]
brk> meth:= ApplicableMethod( IsConfluent, [ col ], 1, "all" );;
[...]
brk> meth[1]( col );
false
brk> meth[2]( col );
true
```

**Comment by hulpke** [comment]
I think the issue is that this is not a polycyclic presentation. (I seem to recall that it is not possible to create the trivial group as a pc group, and there is actually some workaround in places.)

**Comment by ThomasBreuer** [comment]
@hulpke Yes. Still it is irritating that the two methods disagree on whether or not the given rws is confluent or not.

(If the rws is confluent then creating a pc group from it should be not too expensive.
The documentation is not really straightforward: The manual section "Constructing Pc Groups" warns that in general  `PcGroupFpGroup` cannot be used, but I find the hint "see Chapter 47.14 for further information" not helpful; this is the section "Quotient Methods".)

**Comment by hulpke** [comment]
i'm not sure I understand what you want the function to do. `PcGroupFpGroup` is a converter function that builds pc groups from a (rather restricted form) finitely presented group. Crucially, it is not a solvable quotient algorithm.
It is there to allow a user to create pc groups without having to start thinking about storage size of words etc.

The internal code for pc groups does not allow for the trivial pc group with no generators, thus the error. 

Is the issue simply about the text of the error message (complaining about confluence)?
Or do you want to test for confluence (and then PcGroupFpGroup is not the right function to use)?

**Comment by ThomasBreuer** [comment]
> So I'd say: both methods for `IsConfluent` probably would be better off if they first checked that all relative orders are non-zero and different from 1, to uniformly reject this example.

Fine.
Then the second `IsConfluent` method in the initial example should be changed. I can make a pull request for that.
